ternary_search/ternary_search.cpp

```diff
@@ -0,0 +1,64 @@
+#include <iostream>
+
+using namespace std;
+/*
+ * Function implementing ternary search
+ * args:
+ *		ar		: Initial array containing elements (unsorted)
+ *		n		: the search tree's depth
+ *		left : left index of array.
+ *		right : right index of array.
+ *		x : element to be searched.
+ * Time Complexity : O(max(input))
+ */  
+int ternary_search (int ar[],int n, int left, int right, int x){
+
+    if(left < 0 || right > n - 1 || left > right){
+        return -1;
+    }
+    
+    if(x == ar[left]){
+        return left;
+    }
+
+    if(x == ar[right]){
+        return right;
+    }
+
+   // Update the two index left and right if the element is not found.
```

Comment:
Please put comments above the relevant blocks and remove unnecessary blank lines.